scopes/typescript/typescript/typescript.main.runtime.ts

```diff
@@ -114,9 +140,18 @@ export class TypescriptMain {
    * add the default package json properties to the component
    * :TODO @gilad why do we need this DSL? can't I just get the args here.
    */
-  getPackageJsonProps(): PackageJsonProps {
+  getCjsPackageJsonProps(): PackageJsonProps {
+    return {
+      main: 'dist/{main}.js',
+      types: '{main}.ts',
+    };
+  }
+
+  getEsmPackageJsonProps(): PackageJsonProps {
     return {
+      // main: 'dist-esm/{main}.js',
       main: 'dist/{main}.js',
+      type: 'module',
```

Comment:
we should be careful with this property, it causes errors, usually when packaging both cjs and mjs in the same package.
I think it's ok if the package is fully esm.

maybe we should set ```type: 'commonjs'``` for cjs? reference - https://nodejs.org/docs/latest-v13.x/api/esm.html#esm_package_json_type_field

Reply:
That's a dedicated API to do exactly this.
it called `getEsmPackageJsonProps` for a reason.. 
it should be used by an env only if you compile to esm..